### PR TITLE
Environmentvars in report

### PR DESCRIPTION
**Pull request by edyoung**
Addresses #2506 

Adds a property `EnvironmentVariables` to the Benchmark JSON output. The value is an object with keys and values for the environment variables, like this:

```json
"EnvironmentVariables":{
    "foo":"bar"
    }
```
If no variables are specified the output includes an empty object, eg

```json
"EnvironmentVariables": {}
```

Should other exporters like XmlExporter be updated for symmetry?

The production code change just in JsonExporterBase. The other changes are for the unit tests.

### CODE REVIEW

File: src/BenchmarkDotNet/Exporters/Json/JsonExporterBase.cs [hunk 1]
```diff
@@ -74,6 +74,7 @@ public override void ExportToLog(Summary summary, ILogger logger)
                 { "Method", report.BenchmarkCase.Descriptor.WorkloadMethod.Name },
                 { "MethodTitle", report.BenchmarkCase.Descriptor.WorkloadMethodDisplayInfo },
                 { "Parameters", report.BenchmarkCase.Parameters.PrintInfo },
+                { "EnvironmentVariables", report.BenchmarkCase.Job.Environment.EnvironmentVariables ?? new List<Jobs.EnvironmentVariable>() },
```
Comment (timcassell) [inline src/BenchmarkDotNet/Exporters/Json/JsonExporterBase.cs:77]: Maybe `.ToDictionary` would make it look better?

Comment (edyoung) [reply src/BenchmarkDotNet/Exporters/Json/JsonExporterBase.cs]: Thanks, I think that is an improvement. .ToDictionary() can throw if the list contains duplicate keys, but BenchmarkDotNet already ensures that the list doesn't contain duplicates elsewhere, so I don't believe we need to account for that here.

**Comment by kant2002** [comment]
@adamsitnik or @AndreyAkinshin what do you think about this PR?

**Comment by AndreyAkinshin** [comment]
@edyoung thank you very much for the PR.

Right now I'm working on a new serialization format that would naturally handle all possible metadata values (including environment variables). I'm also going to provide a set of API for easy manipulation and analysis of the data. I will post updates once I obtain a ready-to-use prototype.

The current json format is going to be deprecated in favor of the new one. I appreciate the effort you put into the PR, but I'm going to close since I feel that it doesn't make sense to extend the existing json format.

**Pull request closed without merging**